src/Integration.Vsix/SonarAnalyzerManager.cs

```diff
@@ -37,52 +44,232 @@ public sealed class SonarAnalyzerManager
+        private ISonarQubeIssuesProvider sonarqubeIssueProvider;
+        private SuppressionHandler suppressionHandler;
+
+        public SonarAnalyzerManager(IServiceProvider serviceProvider, IQualityProfileProvider qualityProfileProvider)
         {
-            if (activeSolutionBoundTracker == null)
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
+            if (qualityProfileProvider == null)
             {
-                throw new ArgumentNullException(nameof(activeSolutionBoundTracker));
+                throw new ArgumentNullException(nameof(qualityProfileProvider));
             }
 
-            if (workspace == null)
+            this.serviceProvider = serviceProvider;
+            this.qualityProfileProvider = qualityProfileProvider;
+
+            this.activeSolutionBoundTracker = serviceProvider.GetMefService<IActiveSolutionBoundTracker>();
+            this.logger = serviceProvider.GetMefService<ILogger>();
+            this.workspace = serviceProvider.GetMefService<VisualStudioWorkspace>();
+            this.sonarQubeService = serviceProvider.GetMefService<ISonarQubeService>();
+
+            if (activeSolutionBoundTracker == null ||
+                logger == null ||
+                workspace == null ||
+                sonarQubeService == null)
             {
-                throw new ArgumentNullException(nameof(workspace));
+                Debug.Fail($@"There was a problem resolving types from MEF catalog:
+- Null IActiveSolutionBoundTracker? {activeSolutionBoundTracker == null}
+- Null ILogger? {logger == null}
+- Null VisualStudioWorkspace? {workspace == null}
+- Null ISonarQubeService? {sonarQubeService == null}");
+                return;
             }
 
-            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
-            this.workspace = workspace;
+            this.activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
+
+            SetupVsixSonarAnalyzerDelegates();
+            RefreshSuppresionHandlingInNuGetAnalyzers();
+            RefreshQualityProfiles();
+        }
 
-            SonarAnalysisContext.ShouldAnalysisBeDisabled = tree => ShouldAnalysisBeDisabledOnTree(tree);
+        public void Dispose()
+        {
+            this.activeSolutionBoundTracker.SolutionBindingChanged -= OnSolutionBindingChanged;
+            CleanupSuppresionHandlingInNuGetAnalyzers();
+            cachedQualityProfiles.Clear();
         }
 
-        private bool ShouldAnalysisBeDisabledOnTree(SyntaxTree tree)
+        private void OnSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
         {
-            if (tree == null)
+            RefreshSuppresionHandlingInNuGetAnalyzers();
+            RefreshQualityProfiles();
+        }
+
+        #region NuGet SonarAnalyzer Delegates
+        private void RefreshSuppresionHandlingInNuGetAnalyzers()
```

Comment:
This method is responsible of handling setting and un-setting the delegates within the NuGet analyzers.